src/inc/FastConv.h

```diff
@@ -0,0 +1,69 @@
+
+#if !defined(__FastConv_HEADER_INCLUDED__)
+#define __FastConv_HEADER_INCLUDED__
+
+#pragma once
+
+#include "ErrorDef.h"
+#include "RingBuffer.h"
+#include <complex>
+#include<Fft.h>
+
+/*! \brief interface for fast convolution
+*/
+class CFastConv
+{
+public:
+    enum ConvCompMode_t
+    {
+        kTimeDomain,
+        kFreqDomain,
+
+        kNumConvCompModes
+    };
+
+    CFastConv(void);
+    virtual ~CFastConv(void);
+
+    /*! initializes the class with the impulse response and the block length
+    \param pfImpulseResponse impulse response samples (mono only)
+    \param iLengthOfIr length of impulse response
+    \param iBlockLength processing block size
+    \return Error_t
+    */
+    Error_t init(float* pfImpulseResponse, int iLengthOfIr, int iBlockLength = 8192);
+
+    /*! resets all internal class members
+    \return Error_t
+    */
+    Error_t reset ();
+
+    /*! computes the output with reverb
+    \param pfOutputBuffer (mono)
+    \param pfInputBuffer (mono)
+    \param iLengthOfBuffers can be anything from 1 sample to 10000000 samples
+    \return Error_t
+    */
+    Error_t process(float* pfOutputBuffer, const float* pfInputBuffer, int iLengthOfBuffers);
+
+    /*! return the 'tail' after processing has finished (identical to feeding in zeros
+    \param pfOutputBuffer (mono)
+    \return Error_t
+    */
+    Error_t flushBuffer(float* pfOutputBuffer);
+
+    void processFrequencyDomain(float *pfOutputBuffer, const float* pfInputBuffer, int iLengthofBuffers);
+
+private:
+    float* m_ImpulseResponse;
+    float* m_InputSignal;
+    int m_IRlen;
+    int m_blockLength;
+    CRingBuffer<float> *m_pcRingBuff;
+    float* pOutputBufferFreq = 0;
+    CFft* fft = 0;
```

Comment:
These two member variables, along with the `processFrequencyDomain`, will not be used in time domain convolution. It should be better to split two types of convolver into two separate classes and add a new base class for them. Then store the pointer of the new base class here. In that case you can use polymorphism to make code clean and split their duties.